chalk-solve/src/clauses.rs

```diff
@@ -454,6 +460,8 @@ fn match_type_name<I: Interner>(builder: &mut ClauseBuilder<'_, I>, name: TypeNa
             .db
             .associated_ty_data(type_id)
             .to_program_clauses(builder),
+        TypeName::Scalar(_) => (),
+        TypeName::Tuple(_) => (),
```

Comment:
I think this is right, though it may change as we tweak our def'n of implied bounds